src/components/common/CreativeCommons.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo } from 'react';
+
+import { Typography } from '@mui/material';
+
+import { DiscriminatedItem } from '@graasp/sdk';
+import {
+  CCSharingVariant,
+  CreativeCommons as GraaspCreativeCommons,
+} from '@graasp/ui';
+
+import { useLibraryTranslation } from '../../config/i18n';
+import LIBRARY from '../../langs/constants';
+
+const convertLicense = (ccLicenseAdaption: string) => {
+  // Legacy licenses.
+  if (['alike', 'allow'].includes(ccLicenseAdaption)) {
+    return {
+      requireAccreditation: true,
+      allowCommercialUse: true,
+      allowSharing: ccLicenseAdaption === 'alike' ? 'alike' : 'yes',
+    };
+  }
+
+  return {
+    requireAccreditation: ccLicenseAdaption?.includes('BY'),
+    allowCommercialUse: !ccLicenseAdaption?.includes('NC'),
+    allowSharing: (() => {
+      if (!ccLicenseAdaption || !ccLicenseAdaption.length) {
+        return '';
+      }
+      if (ccLicenseAdaption?.includes('SA')) {
+        return 'alike';
+      }
+      return ccLicenseAdaption?.includes('ND') ? 'no' : 'yes';
+    })(),
+  };
+};
+
+type Props = {
+  id?: string;
+  ccLicenseAdaption: DiscriminatedItem['settings']['ccLicenseAdaption'];
+  iconSize?: number;
+};
+
+const CreativeCommons = ({
+  id,
+  ccLicenseAdaption,
+  iconSize = 48,
+}: Props): JSX.Element => {
+  const { t } = useLibraryTranslation();
+
+  const { allowSharing, allowCommercialUse, requireAccreditation } = useMemo(
+    () => convertLicense(ccLicenseAdaption ?? ''),
+    [ccLicenseAdaption],
+  );
```

Comment:
Is the memo needed here ? you are computing the values from the function given a prop. Maybe the function should simply be a custom hook. I don't think that there is any thing else that would re-render the component ?